### PR TITLE
Add workflow to publish linux x86_64

### PR DESCRIPTION
**Pull request by VijayanB**
Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

### Description
Download logstash, build plugin, install plugin and
publish logstash-oss-7.13.2-linux-x86_64.tar.gz

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [] New functionality has been documented.
  - [ ] New functionality has documentation added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Build plugin
+        run: |
+          gem build logstash-output-opensearch.gemspec
+
+      - name: Download and extract logstash
+        run: |
+          mkdir -p prepare-install;
+          mv logstash-output-opensearch-*.gem prepare-install/logstash-output-opensearch.gem;
+          wget --no-verbose https://artifacts.elastic.co/downloads/logstash/logstash-oss-7.13.2-linux-x86_64.tar.gz -P prepare-install/;
```
Comment (vamshin) [inline .github/workflows/cd.yml:24]: can logstash oss version be made dynamic to download?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Publish logstash-oss-with-opensearch-plugin
```
Comment (jmazanec15) [inline .github/workflows/cd.yml:1]: Nit: Could cd.yml be renamed to CD.yml? It seems that is the convention for plugins.

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          cd prepare-install;
+          logstash-7.13.2/bin/logstash-plugin install logstash-output-opensearch.gem;
+          logstash-7.13.2/bin/logstash-plugin list;
+          tar -czf logstash-oss-with-opensearch-plugin-7.13.2-linux-x86_64.tar.gz logstash-7.13.2
+
+      - name: Upload logstash-oss-with-opensearch-plugin
+        uses: actions/upload-artifact@v2
+        with:
+          name: "logstash-oss-with-opensearch-plugin-7.13.2-linux-x86_64.tar.gz"
+          path: prepare-install/logstash-oss-with-opensearch-plugin-7.13.2-linux-x86_64.tar.gz
```
Comment (jmazanec15) [inline .github/workflows/cd.yml:40]: For discussion, should the artifact use `x64` as this is what we use for OpenSearch artifacts: https://opensearch.org/downloads.html or do we want to keep consistent with logstash convention of `x86_64`

File: .github/workflows/cd.yml (hunk 2 above)
Comment (VijayanB) [reply .github/workflows/cd.yml]: Ack

File: .github/workflows/cd.yml (hunk 1 above)
Comment (VijayanB) [reply .github/workflows/cd.yml]: Ack

File: .github/workflows/cd.yml (hunk 3 above)
Comment (VijayanB) [reply .github/workflows/cd.yml]: I renamed to our naming convention

**Review by jmazanec15 — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**